### PR TITLE
update(markdown-it-emoji): improve shortcut list usage

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- rewrite default shortcuts to support typed values
- minor styling updating (applied DT defaults)

https://github.com/markdown-it/markdown-it-emoji/blob/master/lib/data/shortcuts.js

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes

### CODE REVIEW

File: types/markdown-it-emoji/index.d.ts [hunk 1]
```diff
@@ -3,16 +3,43 @@
...
+import { PluginSimple, PluginWithOptions } from "markdown-it/lib";
 
 export as namespace markdownitEmoji;
 
 declare namespace markdownitEmoji {
     interface Options {
         defs?: Record<string, string>;
         enabled?: string[];
-        shortcuts?: Record<string, string | string[]>;
+        shortcuts?: Record<Shortcut, string | string[]>;
```
Comment (Shinigami92) [inline types/markdown-it-emoji/index.d.ts:14]: **suggestion:** What about using [LiteralUnion<Shortcut>](https://github.com/microsoft/TypeScript/issues/29729#issuecomment-471566609)?

Comment (peterblazejewicz) [reply types/markdown-it-emoji/index.d.ts:14]: can you be more specific? I was thinking shortcuts are closed list here, thx!

Comment (Shinigami92) [reply types/markdown-it-emoji/index.d.ts:14]: `LiteralUnion` is just a future proof version of an union type so you get auto-completion but also are allowed to insert any other string if needed. So if there are new shortcuts, they can already be used without updating these type. Where if `LiteralUnion` is not used, you getting a compile-time error and need workarounds like `// @ts-expect-error` or `as unknown as Shortcut`

Comment (peterblazejewicz) [reply types/markdown-it-emoji/index.d.ts]: I've amended the code, the 'Shortcut' type itself is now literal union. But I cannot recall how this
is supposed to work as keyed value for `shortcuts` properties (also amended in the update to represents
optional keys for shortcuts)

**Review by Shinigami92 — APPROVED** [review]
Could be improved, but works for now :slightly_smiling_face:

**Comment by peterblazejewicz** [comment]
Ready to merge 💘

**Pull request merged**